### ISSUE TITLE
How to bypass the captcha during a friend request ?

### ISSUE THREAD

**Issue by Ryan4794** [V4NSH4J/discord-mass-DM-GO#892]
Hello, I am developing a mass friend spammer script in Python, but Discord detects it as a script and asks for a captcha even though it's a verified account, and I am using all the headers copied identically from the original request. So, the problem is not there. I have noticed that when I send a friend request with Mass DM Go, it works normally. I have tried to figure out how to bypass the captcha request in the code, but I have never worked with Go, and the script is too complex for me to understand. Therefore, I would like to know how to bypass the captcha request when sending a friend request. Thank you.

**Comment by Ryan4794** [comment]
headers = {
    "accept": "*/*",
    "accept-encoding": "gzip, deflate, br",
    "accept-language": "fr,fr-FR;q=0.9,en;q=0.8,en-GB;q=0.7,en-US;q=0.6",
    "authorization": "MTAwNTUwMTEyMDk1MDU3NTEwNQ.G7IsmA.a58ZVyO727A8xM9EbIlASL_CQAhdIP9iA05JcQ",
    "content-length": "41",
    "content-type": "application/json",
    "cookie": "__dcfduid=d4840c80b91c11edab71fb8ea4259da4; __sdcfduid=d4840c81b91c11edab71fb8ea4259da4284a8b26f6728b4783315467610ed9472f4da9499bafd311e0fa327e6d6eb427; _ga=GA1.2.811230283.1677949339; locale=fr; OptanonConsent=isIABGlobal=false&datestamp=Mon+Mar+06+2023+15%3A26%3A15+GMT%2B0100+(heure+normale+d%E2%80%99Europe+centrale)&version=6.33.0&hosts=&landingPath=NotLandingPage&groups=C0001%3A1%2CC0002%3A0%2CC0003%3A0&AwaitingReconsent=false; __cfruid=257ef69aa2f47dfe579b78972b45d0f5aeb6c565-1678114159; __cf_bm=s7IRgBRgxXPOs0KKxUnaL3ybSgOKRQ0fYZmOm4jtYsQ-1678117489-0-AUPaVRVSuug40RVCGA/pI2mqALYo/39y2FqLxmkvU/UZKuGWZVS9M3am5CA5sP1rP0S0D8GQf8QiIWmnWc5syRgnpYt0ji7GBpuv1eUCmfqwsgnApE7m4GfXMuBbPR82glADNumcI3VMJyovNc/hnJ0=",
    "origin": "https://discord.com",
    "referer": "https://discord.com/channels/@me",
    "sec-ch-ua": "\"Chromium\";v=\"110\", \"Not A(Brand\";v=\"24\", \"Microsoft Edge\";v=\"110\"",
    "sec-ch-ua-mobile": "?0",
    "sec-ch-ua-platform": "\"Windows\"",
    "sec-fetch-dest": "empty",
    "sec-fetch-mode": "cors",
    "sec-fetch-site": "same-origin",
    "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36 Edg/110.0.1587.63",
    'x-context-properties': 'eyJsb2NhdGlvbiI6IkFkZCBGcmllbmQifQ==',
    'x-debug-options': 'bugReporterEnabled',
    'x-discord-locale': 'fr',
    'x-super-properties': 'eyJvcyI6IldpbmRvd3MiLCJicm93c2VyIjoiQ2hyb21lIiwiZGV2aWNlIjoiIiwic3lzdGVtX2xvY2FsZSI6ImZyIiwiYnJvd3Nlcl91c2VyX2FnZW50IjoiTW96aWxsYS81LjAgKFdpbmRvd3MgTlQgMTAuMDsgV2luNjQ7IHg2NCkgQXBwbGVXZWJLaXQvNTM3LjM2IChLSFRNTCwgbGlrZSBHZWNrbykgQ2hyb21lLzExMC4wLjAuMCBTYWZhcmkvNTM3LjM2IEVkZy8xMTAuMC4xNTg3LjYzIiwiYnJvd3Nlcl92ZXJzaW9uIjoiMTEwLjAuMC4wIiwib3NfdmVyc2lvbiI6IjEwIiwicmVmZXJyZXIiOiIiLCJyZWZlcnJpbmdfZG9tYWluIjoiIiwicmVmZXJyZXJfY3VycmVudCI6Imh0dHBzOi8vd3d3LmJpbmcuY29tLyIsInJlZmVycmluZ19kb21haW5fY3VycmVudCI6Ind3dy5iaW5nLmNvbSIsInNlYXJjaF9lbmdpbmVfY3VycmVudCI6ImJpbmciLCJyZWxlYXNlX2NoYW5uZWwiOiJzdGFibGUiLCJjbGllbnRfYnVpbGRfbnVtYmVyIjoxNzg1OTAsImNsaWVudF9ldmVudF9zb3VyY2UiOm51bGwsImRlc2lnbl9pZCI6MH0='
    }

**Comment by SubnauticaNitroxx** [comment]
You ever figure this out?

**Comment by Ryan4794** [comment]
Yes, you just need to have good x-super-properties and all the right headers (x-super-properties is a base64 encoded header), you also need to use a good client like tls-client (py)


```
ja3_string = "771,4865-4866-4867-49195-49199-49196-49200-52393-52392-49171-49172-156-157-47-53,0-23-65281-10-11-35-16-5-13-18-51-45-43-27-17513,29-23-24,0"
VERSION = 131

def getsession():
    session = tls_client.Session(
        client_identifier=f"chrome_{VERSION}",
        random_tls_extension_order=True,
        #ja3_string=ja3_string
    )
    return session
    
    
DEFBUILDNUMBER = 348191
    
def getx():    # get x-super-properties
    data = {'os': 'Windows', 'browser': 'Chrome', 'device': '', 'system_locale': 'fr-FR', 'browser_user_agent': userant, 'browser_version': f'{VERSION}.0.0.0', 'os_version': '10', 'referrer': 'https://discord.com/', 'referring_domain': 'discord.com', 'referrer_current': '', 'referring_domain_current': '', 'release_channel': 'stable', 'client_build_number': DEFBUILDNUMBER, 'client_event_source': None}



    json_data = json.dumps(data, separators=(',', ':'))
    data = base64.b64encode(json_data.encode()).decode()

    return data    
    
    
session = getsession()



session.headers = {
            "accept": "*/*",
            "accept-encoding": "gzip, deflate, br, zstd",
            "accept-language": "fr-FR,fr;q=0.9,en-US;q=0.8,en;q=0.7",
            #"cache-control": "no-cache",
  
            "origin": "https://discord.com", 
            #"pragma": "no-cache",
            "priority": "u=1, i",
            "referer": "https://discord.com/channels/@me",
            "sec-ch-ua": f'Google Chrome";v="{VERSION}", "Chromium";v="{VERSION}", "Not_A Brand";v="24',


            "sec-ch-ua-mobile": "?0",
            "sec-ch-ua-platform": '"Windows"',
            "sec-fetch-dest": "empty",
            "sec-fetch-mode": "cors",
            "sec-fetch-site": "same-origin",

            "user-agent": userant,

            "x-debug-options": "bugReporterEnabled",
            "x-discord-locale": "fr",
            "x-discord-timezone": "Europe/Brussels",
            "x-super-properties": getx()
        }




payload = {"data":  "blablabla"}



r = session.post("https://discord/api/blablabla", json=payload)
```



PS: 'x-context-properties': 'eyJsb2NhdGlvbiI6IkFkZCBGcmllbmQifQ==', is also important, it is base64 encoded like the x super propeties
